src/Campaigns/Blocks/CampaignStats/render.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use Give\Campaigns\CampaignDonationQuery;
+use Give\Campaigns\Models\Campaign;
+use Give\Campaigns\Repositories\CampaignRepository;
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Framework\Support\ValueObjects\Money;
+
+/**
+ * @unreleased
+ *
+ * @var array    $attributes
+ * @var Campaign $campaign
+ */
+
+if (
+    !isset($attributes['campaignId'])
+    || !$campaign = give(CampaignRepository::class)->getById($attributes['campaignId'])
+) {
+    return;
+}
+
+$query = (new CampaignDonationQuery($campaign))
+    ->select(
+        $attributes['statistic'] === 'top-donation'
+            ? 'MAX(amountMeta.meta_value) as amount'
+            : 'AVG(amountMeta.meta_value) as amount'
+    )
+    ->joinDonationMeta(DonationMetaKeys::AMOUNT, 'amountMeta');
+
+$donationStat = $query->get();
+
+$amount = Money::fromDecimal($donationStat->amount, give_get_currency());
```

Comment:
It would be a good idea to check if `donationStat` actually returned something before using it. If the returned value is `null`, there will be a fatal error.

Reply:
Resolved: d952efdd6c603da1e03badd54aa12d494f2c4b7b